x-pack/solutions/observability/plugins/infra/public/pages/logs/page_content.tsx

```diff
@@ -77,9 +81,35 @@ export const LogsPageContent: React.FunctionComponent = () => {
         <Route
           path="/stream"
           exact
-          render={() => {
-            share.url.locators.get<LogsLocatorParams>(LOGS_LOCATOR_ID)?.navigate({});
-
+          render={({ location }) => {
+            const searchParams = new URLSearchParams(location.search);
+            const g = searchParams.get('_g') || '';
+            const a = searchParams.get('_a') || '';
+
+            const gState = (safeDecode(g) || {}) as unknown as DiscoverGlobalState;
+            const aState = (safeDecode(a) || {}) as unknown as DiscoverAppState;
+
+            const dataViewId =
+              aState.dataSource?.type === 'dataView'
+                ? (aState.dataSource as DataViewDataSource).dataViewId
+                : '';
+
+            const locatorParams = {
```

Comment:
Just a couple of small nits/suggestions:
- This could be typed as `DiscoverAppLocatorParams` if you want better typings here.
- All `DiscoverAppLocatorParams` props are optional, so you could skip the default values here and just rely on Discover defaults if preferred.
- There's also a strongly typed version of the locator available on the Discover start contract that could be used for this.